x-pack/plugins/viz_editor/public/pseudo_plugins/indexpattern_datasource_plugin/indexpattern_datasource_plugin.tsx

```diff
@@ -51,11 +52,46 @@ function DataPanel(props: PanelComponentProps<VisModel>) {
   );
 }
 
+function queryToEsAggsConfigs(query: Query): any {
+  return query.select.reverse().map((selectOperation, index) => {
+    switch (selectOperation.operation) {
+      case 'count':
+        return { enabled: true, id: String(index), params: {}, schema: 'metric', type: 'count' };
+      case 'terms':
+        return {
+          id: String(index),
+          enabled: true,
+          type: 'terms',
+          schema: 'segment',
+          params: {
+            field: selectOperation.argument.field,
+            size: selectOperation.argument.size,
+          },
+        };
+      case 'date_histogram':
+        return {
+          id: String(index),
+          enabled: true,
+          type: 'date_histogram',
+          schema: 'segment',
+          params: {
+            field: selectOperation.argument.field,
+            interval: 'd',
+          },
+        };
+    }
+  });
+}
+
 function toExpression(viewState: VisModel) {
+  if (Object.keys(viewState.queries).length === 0) {
+    return '';
+  }
+  const firstQuery = Object.values(viewState.queries)[0];
```

Comment:
We're ignoring subsequent queries? I vote that we make our view model singular, then, as otherwise, it's misleading, and might be hard to track down all of the places where it doesn't do what we expect.

Reply:
You are right, I just didn't get around to completely remove this. I also feel like we can omit it for now - ideally the editor plugin wouldn't even need to know about multiple queries and it's handled completely inside the datasource plugin (e.g. the query is split up into multiple `esaggs` if necessary). I would like to explore this more, but IMHO this can wait till after our MVP

Reply:
I agree that this can wait, but maybe we could throw an error here in case there are multiple queries?

Reply:
Makes sense - I moved the error to the root state reducer in `main.tsx`